dropwizard-auth/src/main/java/io/dropwizard/auth/PolymorphicAuthValueFactoryProvider.java

```diff
@@ -0,0 +1,133 @@
+package io.dropwizard.auth;
+
+import org.glassfish.hk2.api.InjectionResolver;
+import org.glassfish.hk2.api.ServiceLocator;
+import org.glassfish.hk2.api.TypeLiteral;
+import org.glassfish.hk2.utilities.binding.AbstractBinder;
+import org.glassfish.jersey.server.internal.inject.AbstractContainerRequestValueFactory;
+import org.glassfish.jersey.server.internal.inject.AbstractValueFactoryProvider;
+import org.glassfish.jersey.server.internal.inject.MultivaluedParameterExtractorProvider;
+import org.glassfish.jersey.server.internal.inject.ParamInjectionResolver;
+import org.glassfish.jersey.server.model.Parameter;
+import org.glassfish.jersey.server.spi.internal.ValueFactoryProvider;
+
+import java.util.Set;
+import javax.inject.Inject;
+import javax.inject.Singleton;
+import java.security.Principal;
+
+/**
+ * Value factory provider supporting injection of a hierarchy of
+ * {@link Principal} subclasses by the {@link Auth} annotation.
+ *
+ * @param <T> the type acting as the superclass from which injected
+ *            principals inherit
+ */
+@Singleton
+public class PolymorphicAuthValueFactoryProvider<T extends Principal>
+    extends AbstractValueFactoryProvider
+{
+    /**
+     * Set of provided {@link Principal} subclasses.
+     */
+    protected final Set<Class<? extends T>> principalClassSet;
+
+
+    /**
+     * {@link Principal} value factory provider injection constructor.
+     *
+     * @param mpep                   multivalued parameter extractor provider
+     * @param injector               injector instance
+     * @param principalClassProvider provider of the principal class
```

Comment:
The parameter name in the docs doesn't match the actual parameter name `principalClassSetProvider`